app/services/group_stage_service.rb

```diff
@@ -0,0 +1,37 @@
+# frozen_string_literal: true
+
+class GroupStageService
+  def self.generate_group_stage(groups)
+    return false if groups.length.zero?
+
+    average_group_size = (groups.flatten.length.to_f / groups.length.to_f)
+    if (average_group_size % 1).zero?
+      groups = groups.map { |group| get_group_object_from(group) }
+      group_stage = Stage.new level: -1, groups: groups
+      group_stage
+    else
+      false
+    end
+  end
+
+  def self.get_group_object_from(team_array)
+    matches = generate_all_matches_between team_array
```

Comment:
unnecessary var-assignment again, you're wasting our ram /o\

Reply:
I don't think those are unnecessary. For computing purposes they are unnecessary, but i think it helps quite a bit with readability.